include/glow/Graph/Graph.h

```diff
@@ -460,23 +460,44 @@ class Function final : public Named {
+  /// individual quantization parameters for each bias element corresponding to
+  /// each output channel. \p kernels defines the size of the height and width
   /// dimensions of the filters. \p strides defines the number of steps to take
   /// in the input for each output cell. \p pads defines how many zero padding
   /// cells should be added to the input during convolution. \p group defines
   /// the number of groups the input and output channels should be divided into
-  /// and convolved separately. If bias is FloatTy then it will be quantized
-  /// to Int32QTy automatically.
-  /// NOTE: ChannelwiseQuantizedConvolutionNode does
-  /// not yet have an implementation so attempting to run a graph containing
-  /// this node fails.
+  /// and convolved separately. \p dilation defines the filter dilation.
+  /// This function is flexible and has the following features:
+  /// - it can be provided with a floating-point \p filter and the function will
+  ///   quantize automatically the filter channelwise using the given schema
+  ///   \p schema and type \p filterElemQTy.
+  /// - it can be provided with a floating-point \p bias and the function will
+  ///   quantize automatically the bias channelwise using the given schema
+  ///   \p schema and type \p biasElemQTy.
+  /// - if \p filter is floating-point and \p filterScales or \p filterOffsets
+  ///   are not provided then this function will derive them automatically.
+  /// - if \p filter is quantized then \p filterScales or \p filterOffsets are
+  ///   mandatory.
+  /// - if \p bias is floating-point and \p biasScales or \p biasOffsets are not
+  ///   provided then this function will derive them automatically.
+  /// - if \p bias is quantized  and \p biasScales or \p biasOffsets are not
+  ///   provided then this function will assume the implicit parameters
+  ///   biasScales[i] = inputScale * filterScales[i] and biasOffsets[i] = 0.
+  ///   To be noted that this case can handle safely only INT32 bias data type
+  ///   because for INT8 type the bias will almost certainly be saturated.
+  /// This function requires \p filter and \p bias operands to be constants.
   ChannelwiseQuantizedConvolutionNode *createChannelwiseQuantizedConv(
       llvm::StringRef name, NodeValue input, NodeValue filter, NodeValue bias,
-      NodeValue scales, NodeValue offsets, TypeRef outTy,
-      llvm::ArrayRef<unsigned_t> kernels, llvm::ArrayRef<unsigned_t> strides,
-      llvm::ArrayRef<unsigned_t> pads, unsigned_t group);
+      NodeValue filterScales, NodeValue filterOffsets, NodeValue biasScales,
+      NodeValue biasOffsets, TypeRef outTy, llvm::ArrayRef<unsigned_t> kernels,
+      llvm::ArrayRef<unsigned_t> strides, llvm::ArrayRef<unsigned_t> pads,
+      unsigned_t group, unsigned_t dilation = 1,
+      quantization::Schema schema = quantization::Schema::Asymmetric,
+      ElemKind filterElemQTy = ElemKind::Int8QTy,
+      ElemKind biasElemQTy = ElemKind::Int32QTy);
```

Comment:
The reason why I added support for bias types other than Int32QTy is the same reason why I added support for other nodes as well (like standard Convolution and FullyConnected):
- Multiple times in a row we were given a third party library with optimized implementations for quantized kernels for which **we had no control over** (we were only given an API and a library in binary format) and we had to integrate them in Glow. For some reason, the guys who implemented the kernels chose to use Int8 bias. I`m not saying this is a good choice (using int8 bias instead of int32 bias has almost zero gain in terms of memory usage) but we had to do something in order to make Glow capable to prepare data for such kernels.
- Another case when we had to do this is when we used Glow to dispatch (offload) work to an accelerator through a firmware API. Again, that accelerator chose Int8 bias for its quantized implementations. Again we had no control over those implementations.
- I like to think about Glow as a tool generic enough to allow integrating kernels with all sorts of constraints. This is the reason why I also added a long time ago the quantization schema **SymmetricWithPower2Scale** which prepared the data for some quantized kernels which only used a bias up-shift and an output down-shift.